src/Namespace.elm

```diff
@@ -0,0 +1,81 @@
+module Namespace exposing (..)
+
+import Definition exposing (Definition)
+import Json.Decode as Json exposing (andThen, field)
+import List.Nonempty
+import UnisonHash exposing (UnisonHash(..))
+
+
+type alias Path =
+    List.Nonempty.Nonempty String
+
+
+type NamespaceChild
+    = SubNamespace String Int
+    | Type String UnisonHash
+
+
+type alias Namespace =
+    { path : Path, hash : UnisonHash, children : List NamespaceChild }
```

Comment:
`path` currently isn't part of the API so there is a bit of speculation on my part in that area.